File: pset6/credit/credit.py
from cs50 import get_string

number = get_string("Number: ")

if len(number) < 13 or len(number) > 16:
    print("INVALID")
else:
    l = len(number)
    suma_d = 0
    suma_i = 0
    for x in range(0, l, 2):
        if l % 2 == 0:
            num = int(number[x]) * 2
            if num > 9:
                num = 1 + num % 10
            suma_d += num
            suma_i += int(number[x + 1])
        else:
            if x == 0:
                suma_i += int(number[x])
            else:
                num = int(number[x - 1]) * 2
                if num > 9:
                    num = 1 + num % 10
                suma_d += num
                suma_i += int(number[x])

    suma = suma_d + suma_i
    d_two = int(number[1])
    d = int(number[0])

    if (suma % 10 == 0):
        if (d == 3 and l == 15 and (d_two == 3 or d_two == 7)):
            print("AMEX")
        elif (d == 5 and l == 16 and (d_two >= 1 or d_two <= 5)):
            print("MASTERCARD")
        elif (d == 4 and (l == 13 or l == 16)):
            print("VISA")
        else:
            print("INVALID")
    else:
        print("INVALID")

    # print("len: {}, suma_d: {}, suma_i: {}, suma: {}, d: {}, d_two: {}".format(l, suma_d, suma_i, suma, d, d_two))

File: pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Ask user for credit card number
    long number;
    int r = 0;
    number = get_long("Number: ");
    long nro = number;
    for (; nro > 0; nro /= 10)
    {
        r++;
    }


    if (r > 16 || r < 13)
    {
        printf("INVALID\n");
    }
    else
    {
        int d;
        int n = 0;
        int suma_p = 0;
        int suma_i = 0;
        int d_two = 0;
        for (nro = number; nro > 10; nro /= 10)
        {
            d_two = nro % 10;
        }
        for (; number > 0; number /= 10)
        {
            d = number % 10;
            if (n % 2 != 0)
            {
                int num = d * 2;
                if (num >= 10)
                {
                    num = 1 + num % 10;
                }
                suma_p += num;
            }
            else
            {
                suma_i += d;
            }
            n++;
        }
        int sum = suma_p + suma_i;
        if (sum % 10 == 0)
        {
            if (d == 3 && r == 15 && (d_two == 3 || d_two == 7))
            {
                printf("AMEX\n");
            }
            else if (d == 5 && r == 16 && d_two >= 1 && d_two <= 5)
            {
                printf("MASTERCARD\n");
            }
            else if (d == 4 && (r == 13 || r == 16))
            {
                printf("VISA\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else
        {
            printf("INVALID\n");
        }
        //printf("r = %d d = %d, sum = %d sum_p = %d sum_i = %d\n",r,d,sum,suma_p,suma_i);
    }

}